### PR TITLE
Enable server-side rendering for amp-audio

### PR DESCRIPTION
**Pull request by deepaklalwani97**
Fixes https://github.com/ampproject/amphtml/issues/29489
- [x] Allow audio tag without src inside the amp-audio component
- [x] Allow the container layout for the amp-audio component.
- [x] Remove layout logic specific to <amp-audio>, independently of the items above.
- [x] Add default height for the amp-audio component in case of audio tag missing.
- [x] Unit test cases

### CODE REVIEW

File: extensions/amp-audio/0.1/amp-audio.js [hunk 1]
```diff
@@ -141,14 +147,19 @@ export class AmpAudio extends AMP.BaseElement {
       audio
     );
 
-    applyFillContent(audio);
```
Comment (dethstrobe) [inline extensions/amp-audio/0.1/amp-audio.js:144]: Don't we want this still? It styles the audio element with block and a few other things I think we still want. Though I do see we're overriding the width on line 122/128.

but if it is no longer giving us styles we care about, I'm also fine removing it. I just don't have enough historical context to understand why it was here in the first place.

File: extensions/amp-audio/0.1/amp-audio.js [hunk 2]
```diff
@@ -104,7 +100,13 @@ export class AmpAudio extends AMP.BaseElement {
    * Builds the internal <audio> element.
    */
   buildAudioElement() {
-    const audio = this.element.ownerDocument.createElement('audio');
+    let audio = this.element.getElementsByTagName('audio')[0];
```
Comment (jridgewell) [inline extensions/amp-audio/0.1/amp-audio.js:103]: Why not just `querySelector`?

File: extensions/amp-audio/0.1/amp-audio.js [hunk 3]
```diff
@@ -141,14 +143,19 @@ export class AmpAudio extends AMP.BaseElement {
...
+        return;
+      }
       if (child.getAttribute && child.getAttribute('src')) {
         assertHttpsUrl(child.getAttribute('src'), dev().assertElement(child));
       }
       audio.appendChild(child);
     });
-    this.element.appendChild(audio);
+
+    if (!hasAudioTag) {
```
Comment (jridgewell) [inline extensions/amp-audio/0.1/amp-audio.js:156]: I think we can just move this up to the point we create it, no?

File: extensions/amp-audio/0.1/amp-audio.js (hunk 1 above)
Comment (jridgewell) [reply extensions/amp-audio/0.1/amp-audio.js:144]: I don't think we should change this in this PR.

File: src/core/static-layout.js [hunk 4]
```diff
@@ -57,25 +57,7 @@ export function hasNaturalDimensions(tagName) {
 export function getNaturalDimensions(element) {
   const tagName = element.tagName.toUpperCase();
   devAssert(naturalDimensions_[tagName] !== undefined);
-  if (!naturalDimensions_[tagName]) {
```
Comment (jridgewell) [inline src/core/static-layout.js:60]: Ditto, I don't think we should change this in this PR.

File: src/core/static-layout.js [hunk 5]
```diff
@@ -31,7 +31,7 @@ export const naturalDimensions_ = {
   'AMP-PIXEL': {width: '0px', height: '0px'},
   'AMP-ANALYTICS': {width: '1px', height: '1px'},
   // TODO(dvoytenko): audio should have width:auto.
```
Comment (MichaelRybak) [inline src/core/static-layout.js:33]: @deepaklalwani97 Please remove the TODO.

@dvoytenko Since the TODO is assigned to you, do you have any thoughts on the PR?

File: validator/validator-main.protoascii [hunk 6]
```diff
@@ -2048,6 +2048,14 @@ tags: {
   }
   spec_url: "https://amp.dev/documentation/components/amp-audio/"
 }
+tags: {
```
Comment (MichaelRybak) [inline validator/validator-main.protoascii:2051]: Please update the comment in line 2029.

**Review by MichaelRybak — CHANGES_REQUESTED** [review]
Hi. I'll review the validator changes on behalf of wg-caching group.

**Comment by deepaklalwani97** [comment]
Thanks, @jridgewell and @MichaelRybak for the feedback I've addressed them all.

File: extensions/amp-audio/0.1/amp-audio.js [hunk 7]
```diff
@@ -143,12 +148,15 @@ export class AmpAudio extends AMP.BaseElement {
 
     applyFillContent(audio);
     realChildNodes(this.element).forEach((child) => {
+      if (child.tagName === 'AUDIO') {
```
Comment (jridgewell) [inline extensions/amp-audio/0.1/amp-audio.js:151]: Nit: Could this be
```suggestion
      if (child === audio) {
```

File: src/core/static-layout.js [hunk 8]
```diff
@@ -30,8 +30,7 @@ import {setStyle, setStyles, toggle} from '#core/dom/style';
 export const naturalDimensions_ = {
   'AMP-PIXEL': {width: '0px', height: '0px'},
   'AMP-ANALYTICS': {width: '1px', height: '1px'},
-  // TODO(dvoytenko): audio should have width:auto.
-  'AMP-AUDIO': null,
+  'AMP-AUDIO': {width: 'auto', height: '3.5rem'},
```
Comment (jridgewell) [inline src/core/static-layout.js:33]: Don't we still need to undo this change?

**Review by jridgewell — APPROVED** [review]

**Review by MichaelRybak — APPROVED** [review]
Thanks @deepaklalwani97 , approving changes that addressed my comments.

File: extensions/amp-audio/0.1/amp-audio.js (hunk 7 above)
Comment (deepaklalwani97) [reply extensions/amp-audio/0.1/amp-audio.js]: Thanks, @jridgewell I have addressed the changes.
